drift-protocol/src/main/java/com/facebook/drift/protocol/TCompactProtocol.java

```diff
@@ -774,6 +774,16 @@ public ByteBuffer readBinary()
         return ByteBuffer.wrap(buf);
     }
 
+    /**
+     * Read data from wire directly into the buffer
+     */
+    @Override
+    public void readBinary(byte[] buf, int offset, int length)
+            throws TException
+    {
+        transport.read(buf, offset, length);
```

Comment:
Is this correct implementation? https://github.com/prestodb/drift/pull/18/files#diff-6319b225b93d4aa6c903d9d874a7d971R767 works a little different?

Also I would recommend you reading in chunks here as well.

Reply:
Done. Added call to checkSize and reading in chunks.